ribs/emitters/opt/_lm_ma_es.py

```diff
@@ -0,0 +1,243 @@
+"""Implementation of LM-MA-ES that can be used across various emitters.
+
+Adapted from Nikolaus Hansen's pycma:
+https://github.com/CMA-ES/pycma/blob/master/cma/purecma.py
+"""
+import numba as nb
+import numpy as np
+from threadpoolctl import threadpool_limits
+
+from ribs.emitters.opt._evolution_strategy_base import EvolutionStrategyBase
+
+
+class LMMAEvolutionStrategy(EvolutionStrategyBase):
+    """LM-MA-ES optimizer for use with emitters.
+
+    Refer to :class:`EvolutionStrategyBase` for usage instruction.
+
+    Args:
+        sigma0 (float): Initial step size.
+        batch_size (int): Number of solutions to evaluate at a time. If None, we
+            calculate a default batch size based on solution_dim.
+        solution_dim (int): Size of the solution space.
+        seed (int): Seed for the random number generator.
+        dtype (str or data-type): Data type of solutions.
+        weight_rule (str): Method for generating weights. Either "truncation"
```

Comment:
I think we should leave out weight_rule here -- I'm not actually sure LM-MA-ES has an active variant, although there is probably some way to adapt active CMA-ES to active LM-MA-ES. Let's just take out the `weight_rule` argument from this class for now.